frontend/src/scenes/insights/utils/compareFilters.ts

```diff
@@ -4,23 +4,52 @@ import { objectCleanWithEmpty, objectsEqual } from 'lib/utils'
 import { cleanFilters } from './cleanFilters'
 
 /** clean filters so that we can check for semantic equality with a deep equality check */
-const clean = (f: Partial<AnyFilterType>): Partial<AnyFilterType> => {
+const clean = (
+    f: Partial<AnyFilterType>,
+    test_account_filters_default_checked: boolean | undefined
+): Partial<AnyFilterType> => {
     // remove undefined values, empty array and empty objects
-    const cleanedFilters = objectCleanWithEmpty(cleanFilters(f))
+    const cleanedFilters = objectCleanWithEmpty(cleanFilters(f, test_account_filters_default_checked))
+
+    // do we need an order property on events or actions?
+    const needsOrder = (cleanedFilters.events || []).length + (cleanedFilters.actions || []).length > 1
 
     cleanedFilters.events = cleanedFilters.events?.map((e) => {
         // event math `total` is the default
         if (e.math === 'total') {
             delete e.math
         }
+
+        // delete unnecessary event order
+        if (!needsOrder) {
+            delete e.order
+        }
+
         return e
     })
+
+    cleanedFilters.actions = cleanedFilters.actions?.map((a) => {
+        // delete unnecessary action order
+        if (!needsOrder) {
+            delete a.order
+        }
+
+        return a
+    })
```

Comment:
Nit: given these are in-place operations, `.forEach()` or `for (const a of ...)` would better reflect what's happening (i.e. `return a` doesn't matter here)

Reply:
Good catch! I think little things like this make the code much more clear for the next person going over it and have replaced `.map` with `.forEach`.